src/getDefaultMiddleware.ts

```diff
@@ -28,6 +29,24 @@ interface GetDefaultMiddlewareOptions {
   serializableCheck?: boolean | SerializableStateInvariantMiddlewareOptions
 }
 
+export type DefaultMiddlewareResult<
+  S,
+  O extends GetDefaultMiddlewareOptions
+> = Unshift<
```

Comment:
*whimper* :)

Reply:
Yeah, wanna REALLY hurt? Take a look at the implementation of Unshift ;)

Here, I had the choice between
* make `getDefaultMiddleware` always return `[ThunkMiddleware]`, which is untrue
* create a type with one case for every possible combination of `thunk`, `immutableCheck` and `serializableCheck` (I had this in for a while)
* this one: create the tuple of length 0 to 3, and one check per possible configuration option. The only way to make a tuple longer in TS is currently to **prepend** values, to I'm starting with the empty tuple `[]` and prepending the three middlewares depending on their configuration value from the innermost to the outermost `Unshift`. At least it doesn't repeat itself too much :man_shrugging:

Reply:
So now what happens if we add another entry to `getDefaultMiddleware`?

Reply:
This way: another level of `Unshift`, essentially two more lines. I might still extract the types in there out, that will make it more readable.

The way I had it before? Go from `3!` cases to `4!` cases. 

Problem is: now that the `Dispatch` type relies on the value of `middleware`, `getDefaultMiddleware` needs to return it's value accurately.

Although I can think of a fourth way right now: instead of the tuple, it could also just return an array. That might be simpler to type - it was just not possible with an earlier implementation of `DispatchForMiddlewares`. I'll investigate that tomorrow.

Reply:
Okay, it worked out - that's much simpler now.